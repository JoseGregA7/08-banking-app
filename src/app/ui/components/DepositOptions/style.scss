.DepositOptions__wrapper {
  background-color: #f9f9f9;
  padding: 10px;
  border-radius: 8px;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
  max-width: 400px;
  margin: 0 auto;
}

.DepositOptions__title {
  font-size: 1rem;
  font-weight: bold;
  text-align: center;
  margin-bottom: 8px;
  color: #333;
}

.DepositOptions__form {
  display: flex;
  flex-direction: column;
}

.DepositOptions__field {
  margin-bottom: 10px;
}

.DepositOptions__label {
  font-size: 0.8rem;
  font-weight: 600;
  margin-bottom: 5px;
  color: #444;
}

.DepositOptions__select,
.DepositOptions__input {
  padding: 4px;
  font-size: 0.8rem;
  border: 1px solid #ddd;
  border-radius: 4px;
  width: 100%;
  box-sizing: border-box;
  transition: all 0.3s ease;
}

.DepositOptions__select:focus,
.DepositOptions__input:focus {
  outline: none;
  border-color: #007BFF;
  box-shadow: 0 0 5px rgba(0, 123, 255, 0.2);
}

.DepositOptions__submitButton {
  padding: 8px;
  font-size: 0.8rem;
  background-color: #007BFF;
  color: white;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

.DepositOptions__submitButton:hover {
  background-color: #910910;
}

.DepositOptions__submitButton:disabled {
  background-color: #c0c0c0;
  cursor: not-allowed;
}

.DepositOptions__input[type="number"] {
  -moz-appearance: textfield;
  -webkit-appearance: none;
  appearance: none;
}

.DepositOptions__input[type="number"]::-webkit-outer-spin-button,
.DepositOptions__input[type="number"]::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}