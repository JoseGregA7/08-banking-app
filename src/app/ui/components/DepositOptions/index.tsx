import './style.scss';

const DepositOptions = ({
  handleDepositTypeChange,
  handleDepositAmountChange,
  handleDepositSubmit,
  state,
}: {
  handleDepositTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
  handleDepositAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  handleDepositSubmit: (e: React.FormEvent) => void;
  state: any;
}) => {
  return (
    <div className="DepositOptions__wrapper" data-testid="deposit-options-wrapper">
      <h3 className="DepositOptions__title" id="deposit-form-title">
        Formulario de Depósito
      </h3>
      <form
        onSubmit={handleDepositSubmit}
        className="DepositOptions__form"
        aria-labelledby="deposit-form-title"
        data-testid="deposit-form"

      >
        <div className="DepositOptions__field">
          <label htmlFor="depositType" className="DepositOptions__label">
            Tipo de Depósito:
          </label>
          <select
            id="depositType"
            value={state.depositType}
            onChange={handleDepositTypeChange}
            className="DepositOptions__select"
            aria-label="Seleccionar el tipo de depósito"
            data-testid="deposit-type-select"
          >
            <option value="BRANCH">Depósito desde Sucursal</option>
            <option value="ATM">Depósito desde Cajero</option>
            <option value="ANOTHER_ACCOUNT">Depósito desde Otra Cuenta</option>
          </select>
        </div>
        <div className="DepositOptions__field">
          <label htmlFor="amount" className="DepositOptions__label">
            Cantidad a depositar:
          </label>
          <input
            id="amount"
            type="number"
            value={state.depositAmount}
            onChange={handleDepositAmountChange}
            min="0"
            className="DepositOptions__input"
            aria-label="Cantidad a depositar"
            data-testid="deposit-amount-input"
          />
        </div>
        <button
          type="submit"
          className="DepositOptions__submitButton"
          aria-label="Realizar depósito"
          data-testid="deposit-submit-button"
        >
          Realizar Depósito
        </button>
      </form>
    </div>
  );
};

export default DepositOptions;
