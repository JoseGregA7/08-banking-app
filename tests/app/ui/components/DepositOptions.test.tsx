import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { describe, expect, test, vitest } from 'vitest';
import DepositOptions from '../../../../src/app/ui/components/DepositOptions';

describe('DepositOptions', () => {
    const mockHandleDepositTypeChange = vitest.fn();
    const mockHandleDepositAmountChange = vitest.fn();
    const mockHandleDepositSubmit = vitest.fn();
    const state = {
        depositType: 'BRANCH',
        depositAmount: 100,
    };

    test('should call handleDepositSubmit when submit button is clicked', () => {
        const { getByTestId } = render(
          <DepositOptions
            handleDepositTypeChange={mockHandleDepositTypeChange}
            handleDepositAmountChange={mockHandleDepositAmountChange}
            handleDepositSubmit={mockHandleDepositSubmit}
            state={state}
          />
        );
        const formElement = getByTestId('deposit-form');
        expect(formElement).toBeInTheDocument();
        fireEvent.click(getByTestId('deposit-submit-button'));
        expect(mockHandleDepositSubmit).toHaveBeenCalledTimes(1);
      });
      
    test('should call handleDepositTypeChange when deposit type is selected', () => {
        const { getByTestId } = render(
            <DepositOptions
                handleDepositTypeChange={mockHandleDepositTypeChange}
                handleDepositAmountChange={mockHandleDepositAmountChange}
                handleDepositSubmit={mockHandleDepositSubmit}
                state={state}
            />
        );

        fireEvent.change(getByTestId('deposit-type-select'), {
            target: { value: 'ATM' },
        });

        expect(mockHandleDepositTypeChange).toHaveBeenCalledTimes(1);
        expect(mockHandleDepositTypeChange).toHaveBeenCalledWith(
            expect.objectContaining({
                target: expect.objectContaining({ value: 'BRANCH' })
            })
        );
    });

    test('should call handleDepositAmountChange when amount is entered', () => {
        const { getByTestId } = render(
            <DepositOptions
                handleDepositTypeChange={mockHandleDepositTypeChange}
                handleDepositAmountChange={mockHandleDepositAmountChange}
                handleDepositSubmit={mockHandleDepositSubmit}
                state={state}
            />
        );

        fireEvent.change(getByTestId('deposit-amount-input'), {
            target: { value: '150' },
        });

        expect(mockHandleDepositAmountChange).toHaveBeenCalledTimes(1);
    });
    test('should call handleDepositSubmit when submit button is clicked', () => {
        const { getByTestId } = render(
            <DepositOptions
                handleDepositTypeChange={mockHandleDepositTypeChange}
                handleDepositAmountChange={mockHandleDepositAmountChange}
                handleDepositSubmit={mockHandleDepositSubmit}
                state={state}
            />
        );
        fireEvent.click(getByTestId('deposit-submit-button'));
        expect(mockHandleDepositSubmit).toHaveBeenCalledTimes(2);

    });

    test('should match snapshot', () => {
        const { asFragment } = render(
            <DepositOptions
                handleDepositTypeChange={mockHandleDepositTypeChange}
                handleDepositAmountChange={mockHandleDepositAmountChange}
                handleDepositSubmit={mockHandleDepositSubmit}
                state={state}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
