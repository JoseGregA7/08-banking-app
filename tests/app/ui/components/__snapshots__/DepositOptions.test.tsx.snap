// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DepositOptions > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="DepositOptions__wrapper"
    data-testid="deposit-options-wrapper"
  >
    <h3
      class="DepositOptions__title"
      id="deposit-form-title"
    >
      Formulario de Depósito
    </h3>
    <form
      aria-labelledby="deposit-form-title"
      class="DepositOptions__form"
      data-testid="deposit-form"
    >
      <div
        class="DepositOptions__field"
      >
        <label
          class="DepositOptions__label"
          for="depositType"
        >
          Tipo de Depósito:
        </label>
        <select
          aria-label="Seleccionar el tipo de depósito"
          class="DepositOptions__select"
          data-testid="deposit-type-select"
          id="depositType"
        >
          <option
            value="BRANCH"
          >
            Depósito desde Sucursal
          </option>
          <option
            value="ATM"
          >
            Depósito desde Cajero
          </option>
          <option
            value="ANOTHER_ACCOUNT"
          >
            Depósito desde Otra Cuenta
          </option>
        </select>
      </div>
      <div
        class="DepositOptions__field"
      >
        <label
          class="DepositOptions__label"
          for="amount"
        >
          Cantidad a depositar:
        </label>
        <input
          aria-label="Cantidad a depositar"
          class="DepositOptions__input"
          data-testid="deposit-amount-input"
          id="amount"
          min="0"
          type="number"
          value="100"
        />
      </div>
      <button
        aria-label="Realizar depósito"
        class="DepositOptions__submitButton"
        data-testid="deposit-submit-button"
        type="submit"
      >
        Realizar Depósito
      </button>
    </form>
  </div>
</DocumentFragment>
`;
